src/main.ts

```diff
@@ -41,14 +41,47 @@ export default function (pluginOptions: string | BannerPluginOptions): any {
           : file[0]
         const filePath: string = resolve(root, outDir, fileName)
 
+        const excludeOptions = pluginConfig.exclude
+        let exclude: boolean = false
+
+        switch (typeof excludeOptions) {
+          case 'undefined':
+            break
+          case 'string':
+            exclude = fileName.includes(excludeOptions)
+            break
+          case 'function':
+            exclude = excludeOptions(fileName)
+            break
+          case 'object':
+            if (!(excludeOptions instanceof RegExp))
+              throw new Error(
+                `[vite-plugin-banner] The "exclude" option must be a string, a function or a regular expression.`
+              )
+            exclude = excludeOptions.test(fileName)
+            break
+          default:
+            throw new Error(
+              `[vite-plugin-banner] The "exclude" option must be a string, a function or a regular expression.`
+            )
+        }
+
         // Only handle matching files
-        if (includeRegexp.test(fileName) && !excludeRegexp.test(fileName)) {
+        if (
+          includeRegexp.test(fileName) &&
+          !excludeRegexp.test(fileName) &&
+          !exclude
+        ) {
           try {
             // Read the content from target file
             let data: string = fs.readFileSync(filePath, {
               encoding: 'utf8',
             })
-
+            if (typeof pluginConfig.content === 'function') {
```

Comment:
Reply [discussion_r1008825066](https://github.com/chengpeiquan/vite-plugin-banner/pull/16#discussion_r1008825066) :

>The `exclude` here is also simply because It will inject the same comments into the css when injecting js causing an error.

I pulled your fork repo to the local to link for debugging, and found that it was caused by a judgment problem in [main.ts](https://github.com/taozhiyu/vite-plugin-banner/blob/main/src/main.ts#L80-L84)  .

When the `content` option is a function, the first file will be processed in this `if` branch, and `pluginConfig.content = replaceText` will set `content` to be a string type, so starting from the second file will not into this branch.

It is recommended to destructure and rename it in this place to make it a separate method that can keep calling:

````diff
  // Get the plugin config
  const pluginConfig: PluginConfig = formatConfig(pluginOptions)
+ const { content: setContent } = pluginConfig
````

Then the judgment of this branch is adjusted to:

````diff
-if (typeof pluginConfig.content === 'function') {
+if (typeof setContent === 'function') {
````

By the way, other related codes are also adjusted:

````diff
-const replaceText = pluginConfig.content(fileName)
+const replaceText = setContent(fileName)
````

The last two `include` statements also need to add the judgment of `typeof pluginConfig.content === 'string'`.

This way, using `function` as the `content` option should work perfectly, no need for the `exclude` option.